arangod/RestHandler/RestReplicationHandler.cpp

```diff
@@ -1329,6 +1329,15 @@ Result RestReplicationHandler::processRestoreCollection(
     }
 #endif
 
+    if (parameters.get(StaticStrings::UsesRevisionsAsDocumentIds).isNone() &&
+        (parameters.get(StaticStrings::SyncByRevision).isNone() ||
+         parameters.get(StaticStrings::SyncByRevision).isTrue())) {
+      // for restored collections that do not have "syncByRevision" nor
+      // "usesRevisionsAsDocumentIds" set, set "usesRevisionsAsDocumentIds"
+      // to true. This allows the usage of revision trees for the collection.
+      toMerge.add(StaticStrings::UsesRevisionsAsDocumentIds, VPackValue(true));
```

Comment:
While it would be possible to add the `syncByRevision` attribute too, the relevant attribute is `usesRevisionsAsDocumentIds`.
I wanted to keep the change minimal for the stable releases, so I only added the one that is absolutely required.